tests/commands/test_bump_command.py

```diff
@@ -831,3 +831,168 @@ def test_bump_manual_version_disallows_prerelease_offset(mocker):
         "--prerelease-offset cannot be combined with MANUAL_VERSION"
     )
     assert expected_error_message in str(excinfo.value)
+
+
+def test_bump_command_prelease_version_type_via_cli(
+    tmp_commitizen_project, mocker: MockFixture
+):
+    # PRERELEASE
+    tmp_version_file = tmp_commitizen_project.join("__version__.py")
+    tmp_version_file.write("0.1.0")
+    tmp_commitizen_cfg_file = tmp_commitizen_project.join("pyproject.toml")
+    tmp_version_file_string = str(tmp_version_file).replace("\\", "/")
+    tmp_commitizen_cfg_file.write(
+        f"{tmp_commitizen_cfg_file.read()}\n"
+        f'version_files = ["{tmp_version_file_string}"]'
+    )
+
+    create_file_and_commit("feat: new user interface")
+
+    testargs = [
+        "cz",
+        "bump",
+        "--prerelease",
+        "alpha",
+        "--yes",
+        "--version-type",
+        "semver",
+    ]
+    mocker.patch.object(sys, "argv", testargs)
+    cli.main()
+
+    tag_exists = git.tag_exist("0.2.0-a0")
+    assert tag_exists is True
+
+    with open(tmp_version_file, "r") as f:
+        assert "0.2.0-a0" in f.read()
+
+    with open(tmp_commitizen_cfg_file, "r") as f:
+        assert "0.2.0-a0" in f.read()
+
+    # PRERELEASE BUMP CREATES VERSION WITHOUT PRERELEASE
+    testargs = ["cz", "bump", "--yes"]
+    mocker.patch.object(sys, "argv", testargs)
+    cli.main()
+
+    tag_exists = git.tag_exist("0.2.0")
+    assert tag_exists is True
+
+    with open(tmp_version_file, "r") as f:
+        assert "0.2.0" in f.read()
+
+    with open(tmp_commitizen_cfg_file, "r") as f:
+        assert "0.2.0" in f.read()
+
+
+def test_bump_command_prelease_version_type_via_config(
+    tmp_commitizen_project, mocker: MockFixture
+):
+    # PRERELEASE
+    tmp_version_file = tmp_commitizen_project.join("__version__.py")
+    tmp_version_file.write("0.1.0")
+    tmp_commitizen_cfg_file = tmp_commitizen_project.join("pyproject.toml")
+    tmp_version_file_string = str(tmp_version_file).replace("\\", "/")
+    tmp_commitizen_cfg_file.write(
+        f"{tmp_commitizen_cfg_file.read()}\n"
+        f'version_files = ["{tmp_version_file_string}"]\n'
+        f'version_type = "semver"'
+    )
+
+    create_file_and_commit("feat: new user interface")
```

Comment:
These setups are similar across different tests. Would it be possible for us to make them pytest fixtures?

Reply:
Ok. I wrote a simple fixture returning a function with parameters  https://github.com/commitizen-tools/commitizen/pull/686/commits/ea7ac936bc201e9090fa2d27cbae232e901fd4c1#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R30